### ISSUE TITLE
Parameters for a view controllers

### ISSUE THREAD

**Issue by metaphore** [czyzby/gdx-lml#33]
Sometimes it need to pass some parameters to new controllers. Like when creating dialog, pass concrete object to display etc. 

My first idea was to obtain my dialog controller instance and pass parameters there. When `InterfaceService#showDialog(Class)` called, it operates with `ViewDialogController` that wraps actual class but doesn't provide access to it.

I've looked into `InterfaceService` code but had no luck to fund any way to solve it. Am I missing anything or there is no any way for parameterizing controllers?

**Comment by metaphore** [comment]
It seems like I've found one dirty solution, I can annotate my dialog controller with both `@ViewDialog` and `@Component` and then inject it anywhere I need.

**Comment by czyzby** [comment]
I'd advise you to share data through services - controllers should be used only to, well, *control* the views: provide necessary rendering data and handle user actions. If you need some mutable state (like - for example - some data that should be displayed in a dialog), create a separate service (annotated with `@Component`) that you can inject and modify.

For example, `@View MenuController` could inject `@Component ErrorService` that holds current error data. Any time an error is thrown, the service is updated and `InterfaceService` is used to show `@ViewDialog ErrorController` that builds its view according to `ErrorService`.

Basically, your gut feeling was right. If you feel that an additional service is an overkill and you want a stateful controller, inject your controller instance and modify it directly.

**Comment by metaphore** [comment]
Alright, thanks. I was thought about dedicated service to pass state or put some flags to execute something in controller upon view creation, but annotating controller with `@Component` and call its methods directly looks much cleaner for now.

**Comment by czyzby** [comment]
Actually, `@ViewDialog` should already make the instance of the class available in the context (controllers are effectively "singletons"), so I don't think you need the extra `@Component` annotation.

**Comment by metaphore** [comment]
Oh, sure. That's even better!
